### PR TITLE
INT-156: Addphoto/edit for JA wikia in mercury when anonymous edit is disabled. 

### PR DESCRIPTION
**Pull request by mari-mcmurtrie**
We should not display edit or upload photo icons on wikias that have anonymous editing disabled. However, in the usecase where a wiki disables anonymous edits while an anonymous user is editing a wiki, we would still need to show the correct error message

### CODE REVIEW

**Comment by mari-mcmurtrie** [comment]
@inez This is front end code for the anonymous edit change. Will you CR? Thank you~!

File: server/views/_layouts/ember-main.hbs [hunk 1]
```diff
@@ -40,6 +40,7 @@
 			M.prop('translations', {{{json (translations wiki.language.user ns='main')}}});
 			M.prop('weppyConfig', {{{json weppyConfig}}});
 			M.prop('wikiaLanguage', '{{{wiki.language.user}}}');
+			M.prop('wikiaDisableAnonymousEditing', {{{wiki.disableAnonymousEditing}}});
```
Comment (rogatty) [inline server/views/_layouts/ember-main.hbs]: There is no need to introduce new variable as there is already `M.provide('wiki', {{{json wiki}}});`.

File: front/scripts/main/components/ArticleWrapperComponent.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ App.ArticleWrapperComponent = Em.Component.extend(App.LanguagesMixin, App.TrackC
 	},
 
 	contributionFeatureEnabled: Em.computed('model.isMainPage', function (): boolean {
-		return !this.get('model.isMainPage') && this.get('isJapaneseWikia');
+		return !this.get('model.isMainPage') && this.get('isJapaneseWikia') && !M.prop('wikiaDisableAnonymousEditing');
```
Comment (rogatty) [inline front/scripts/main/components/ArticleWrapperComponent.ts]: You can use `Em.get(Mercury, 'wiki.wikiaDisableAnonymousEditing')`

File: server/views/_layouts/ember-main.hbs (hunk 1 above)
Comment (mari-mcmurtrie) [inline server/views/_layouts/ember-main.hbs]: Thanks, I changed.

File: front/scripts/main/components/ArticleWrapperComponent.ts (hunk 2 above)
Comment (mari-mcmurtrie) [inline front/scripts/main/components/ArticleWrapperComponent.ts]: changed.

**Comment by mari-mcmurtrie** [comment]
@lizlux @rogatty Will you CR this please? It is very little change. thank you!

File: front/scripts/main/components/ArticleWrapperComponent.ts [hunk 3]
```diff
@@ -67,7 +67,7 @@ App.ArticleWrapperComponent = Em.Component.extend(App.LanguagesMixin, App.TrackC
 	},
 
 	contributionFeatureEnabled: Em.computed('model.isMainPage', function (): boolean {
-		return !this.get('model.isMainPage') && this.get('isJapaneseWikia');
+		return !this.get('model.isMainPage') && this.get('isJapaneseWikia') && !Em.get(Mercury, 'wiki.disableAnonymousEditing');
```
Comment (lizlux) [inline front/scripts/main/components/ArticleWrapperComponent.ts]: This line looks a little long. If it's over 120 characters it should be split up. https://github.com/Wikia/guidelines/blob/master/JavaScript/CodingConventions.md#white-space-guidelines

**Comment by lizlux** [comment]
Just one comment, but LGTM.

**Comment by rogatty** [comment]
What @lizlux said and LGTM

**Comment by Grunny** [comment]
:+1:

**Comment by mari-mcmurtrie** [comment]
@lizlux  @rogatty  @Grunny thank you guys!!!

**Pull request merged**